lib/logstash/inputs/tcp.rb

```diff
@@ -159,16 +159,16 @@ def handle_socket(socket, client_address, client_port, output_queue, codec)
       end
     end
   rescue EOFError
-    @logger.debug? && @logger.debug("Connection closed", :client => socket.peer)
+    @logger.debug? && @logger.debug("Connection closed", :client => peer)
   rescue Errno::ECONNRESET
-    @logger.debug? && @logger.debug("Connection reset by peer", :client => socket.peer)
+    @logger.debug? && @logger.debug("Connection reset by peer", :client => peer)
```

Comment:
why peeer! Oh whyyyy! Actually I do the same hack in the `Connection` class in beats.